### PR TITLE
Refactor Keras Feature Notebooks

### PR DESCRIPTION
**Pull request by lior-dikstein**
## Pull Request Description:

Refactored feature notebooks in Keras. Standardized function usage across notebooks. Updated model examples to use the ImageNet validation set as the representative dataset (except for the pruning and qat notebooks).

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

**Review by elad-c — COMMENTED** [review]
Note that some comments are relevant to multiple similar tutorials

File: tutorials/notebooks/mct_features_notebooks/keras/example_keras_activation_z_score_threshold.ipynb [hunk 1]
```diff
@@ -1,847 +1,777 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "GpEZ2E1qzWl3",
+   "metadata": {
+    "id": "GpEZ2E1qzWl3"
+   },
+   "source": [
+    "We observe only minor improvements when adjusting the Z-score threshold. This pattern is common for most simple models. However, our testing shows that transformer models tend to benefit more from outlier removal. It is advisable to experiment with these parameters if the quantized accuracy is noticeably lower than the float model’s accuracy."
```
[the remark this answers is not in the document]
Comment (lior-dikstein) [reply tutorials/notebooks/mct_features_notebooks/keras/example_keras_activation_z_score_threshold.ipynb:614]: Good idea, but not the scope of this PR
